packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -536,6 +627,173 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
+          if (reader.type === InstanceType.SERVERLESS_V2) {
+            serverlessInHighestTier = true;
+          }
+          highestTierReaders.push(reader);
+        }
+        if (reader.tier <= 1) {
+          noFailoverTierInstances = false;
+        }
+      }
+      const hasOnlyServerlessReaders = hasServerlessReader && !hasProvisionedReader;
+      if (hasOnlyServerlessReaders) {
+        if (noFailoverTierInstances) {
+          Annotations.of(this).addWarning(
+            `Cluster ${this.node.id} only has serverless readers and no reader is in promotion tier 0-1.`+
+            'Serverless readers in promotion tiers >= 2 will NOT scale with the writer, which can lead to '+
+            'availability issues if a failover event occurs. It is recommended that at least one reader '+
+            'has `scaleWithWriter` set to true',
+          );
+        }
+      } else {
+        if (serverlessInHighestTier && highestTier > 1) {
+          Annotations.of(this).addWarning(
+            `There are serverlessV2 readers in tier ${highestTier}. Since there are no instances in a higher tier, `+
+            'any instance in this tier is a failover target. Since this tier is > 1 the serverless reader will not scale '+
+            'with the writer which could lead to availability issues during failover.',
+          );
+        }
+        if (someProvisionedReadersDontMatchWriter.length > 0 && writer.type === InstanceType.PROVISIONED) {
+          Annotations.of(this).addWarning(
+            `There are provisioned readers in the highest promotion tier ${highestTier} that do not have the same `+
+            'InstanceSize as the writer. Any of these instances could be chosen as the new writer in the event '+
+            'of a failover.\n'+
+            `Writer InstanceSize: ${writer.instanceSize}\n`+
+            `Reader InstanceSizes: ${someProvisionedReadersDontMatchWriter.map(reader => reader.instanceSize).join(', ')}`,
+          );
+        }
+      }
+    }
+  }
+
+  /**
+   * Perform validations on the reader instance
+   */
+  private validateReaderInstance(writer: IAuroraClusterInstance, reader: IAuroraClusterInstance): void {
+    if (writer.type === InstanceType.PROVISIONED) {
+      if (reader.type === InstanceType.SERVERLESS_V2) {
+        if (!instanceSizeSupportedByServerlessV2(writer.instanceSize!, this.serverlessV2MaxCapacity)) {
+          Annotations.of(this).addWarning(
+            'For high availability any serverless instances in promotion tiers 0-1 '+
+            'should be able to scale to match the provisioned instance capacity.\n'+
+            `Serverless instance ${reader.node.id} is in promotion tier ${reader.tier},\n`+
+            `But can not scale to match the provisioned writer instance (${writer.instanceSize})`,
+          );
+        }
+      }
+    } else {
+      // TODO: add some info around serverless instance tiers and matching scaling
+      Annotations.of(this).addInfo('...');
```

Comment:
I suppose this is still a work in progress, but calling your attention to this, just in case.

Reply:
ahh forgot about that. I'll fix.

The PR is not a work in progress though it's ready for a final review 🚀

Reply:
removed